orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -196,6 +197,110 @@ class ORTModule(torch.nn.Module):
 
     def __init__(self, module):
         assert isinstance(module, torch.nn.Module), "'module' must be a torch.nn.Module"
+
+        # Create forward dynamically, so each ORTModule instance will have its own copy.
+        # This is needed to be able to copy the forward signatures from the original PyTorch models
+        # and possibly have different signatures for different instances.
+        def _forward(self, *inputs, **kwargs):
+            '''Forward pass starts here and continues at `_ORTModuleFunction.forward`
+
+            ONNX model is exported the first time this method is executed.
+            Next, we build a full training graph with module_gradient_graph_builder.
+            Finally, we instantiate the ONNX Runtime InferenceSession.
+            '''
+            # TODO: using pytorch for evaluation for now. We will use ORT for evaluation latter.
+            if not self._is_training:
+                return self._original_module(*inputs, **kwargs)
+
+            # Exporting module to ONNX for the first time
+            if not self._onnx_training:
+                if not self._device:
+                    self._device = _utils.get_device_from_input_args_kwargs(self._original_module, *inputs, **kwargs)
+                    if not self._device:
+                        raise RuntimeError('A device must be specified in the model or data!')
+                self._get_inference_graph_and_init_gradient_graph_builder(*inputs, **kwargs)
```

Comment:
When we change the signature of the forward method to reflect the signature of the original module, are we still able to access ```*inputs``` and ```**kwargs``` as usual? (since we don't have those inputs anymore?)

Reply:
Yes, the signature is changed only externally. That's how `functools.update_wrapper` works.